smithy-aws-cloudformation/src/main/java/software/amazon/smithy/aws/cloudformation/schema/model/Property.java

```diff
@@ -41,24 +41,30 @@ public final class Property implements ToNode, ToSmithyBuilder<Property> {
     // * writeOnly
 
     private Property(Builder builder) {
-        this.insertionOrder = builder.insertionOrder;
-        this.dependencies = ListUtils.copyOf(builder.dependencies);
-        this.schema = builder.schema;
-    }
+        Schema.Builder schemaBuilder;
 
-    @Override
-    public Node toNode() {
-        ObjectNode.Builder builder = schema.toNode().expectObjectNode().toBuilder();
+        if (builder.schema == null) {
+            schemaBuilder = Schema.builder();
+        } else {
+            schemaBuilder = builder.schema.toBuilder();
+        }
 
-        // Only serialize these properties if set to non-defaults.
-        if (insertionOrder) {
-            builder.withMember("insertionOrder", Node.from(insertionOrder));
+        this.insertionOrder = builder.insertionOrder;
```

Comment:
I initially did that, but kept the existing fields so that we don't need to run the `toNode()` methods each time we access them:
```
        this.insertionOrder = schema.getExtension("insertionOrder")
                .map(n -> n.toNode().expectBooleanNode().getValue()).orElse(false);

        this.dependencies = schema.getExtension("dependencies")
                .map(n -> n.toNode().expectArrayNode().getElementsAs(StringNode::getValue))
                .orElseGet(ArrayList::new);
```
This logic would essentially move into the getters if we got rid of the fields.